services/user.service.js

```diff
@@ -0,0 +1,150 @@
+const filename = process.cwd() + "/data/movies.json";
+const fs = require("fs");
+let movies = require(filename);
+const helper = require("../helpers/helper.js");
+
+/**
+ * This function returns all the movies
+ *
+ * @returns {Array} Returns all the movies
+ */
+function getMovies() {
+  return new Promise((resolve, reject) => {
+    if (movies.length === 0) {
+      reject({
+        message: "No movies available!",
+        status: 202,
+      });
+    }
+    resolve(movies);
+  });
+}
+
+/**
+ * This function returns a movie by given id
+ *
+ * @param {number} id The movie id
+ * @returns {Object} Returns the movie if present
+ */
+function getMovie(id) {
+  return helper.searchByIdInArray(movies, id);
+}
+
+/**
+ * This function inserts a new movie into the JSON file
+ *
+ * @param {Object} newMovie The details about the movie
+ * @returns {Object} Returns the newly created movie
+ */
+function insertMovie(newMovie) {
+  return new Promise((resolve, reject) => {
+    const id = { id: getNewId(movies) };
+    const date = {
+      createdAt: newDate(),
+      updatedAt: newDate(),
+    };
+    newMovie = { ...id, ...newMovie, ...date };
+    movies.push(newMovie);
+    writeJSONFile(movies);
+    resolve(newMovie);
+  });
+}
+
+/**
+ * This function updares a movie by given id and rewrites the JSON file
+ *
+ * @param {number} id The movie id
+ * @param {object} newMovie The details about the updated movie
+ * @returns {Object} Returns the updated movie if updated successfully
+ */
+function updateMovie(id, newMovie) {
+  return helper.searchByIdInArray(movies, id).then((movie) => {
+    const index = movies.findIndex((m) => m.id == movie.id);
+
+    id = { id: movie.id };
+    const date = {
+      createdAt: movie.createdAt,
+      updatedAt: newDate(),
+    };
+
+    movies[index] = { ...id, ...date, ...newMovie };
+    writeJSONFile(movies);
+    resolve(movies[id]);
+  });
+}
+
+/**
+ * This function deletes a movie by given id and rewrites the JSON file
+ *
+ * @param {number} id The movie id
+ * @returns {undefined}
+ */
+function deleteMovie(id) {
+  return helper
+    .searchByIdInArray(movies, Number(id))
```

Comment:
I think you don't need this `searchByIdInArray` call here as you don't use the result. You just need to filter out the deleted movie, update the store and return the id of the deleted movie.